app/src/main/java/com/github/freeman/bootcamp/auth/ProfileCreationActivity.kt

```diff
@@ -0,0 +1,121 @@
+package com.github.freeman.bootcamp.auth
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.github.freeman.bootcamp.R
+import com.github.freeman.bootcamp.ui.theme.BootcampComposeTheme
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+
+/**
+ * The activity that create the GuessIt profile in the real time database
+ * linked with the google profile
+ */
+class ProfileCreationActivity : ComponentActivity() {
+    private val dbref = Firebase.database.getReference("Profiles")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            BootcampComposeTheme {
+                ProfileCreationScreen(dbref)
+            }
+        }
+    }
+}
+
+/**
+ * The bar where the user chooses his/her username
+ */
+@Composable
+fun UsernameBar(
+    username: String,
+    onUsernameChange: (String) -> Unit,
+    onSendClick: () -> Unit
+) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .testTag("usernameBar")
+            .padding(8.dp)
+            .fillMaxSize()
+    ) {
+
+        OutlinedTextField(
```

Comment:
Suggestion: Add limitations to what the user can enter in this field. I was able to write a username that is more than one line long, which was then not added in the database, and I didn't get any error message. This means that a user could make this mistake and not be notified that his profile wasn't recorded.

Reply:
I didn't think about that so thank you for raise this problem.
As I have other features to add in the next sprint, I will keep it in mind and change it in next sprint.